rewrite-java/src/main/java/org/openrewrite/java/format/SpacesVisitor.java

```diff
@@ -91,15 +104,40 @@ <T extends J> JRightPadded<T> spaceBeforeRightPaddedElement(JRightPadded<T> cont
     }
 
     <T extends J> JRightPadded<T> spaceAfter(JRightPadded<T> container, boolean spaceAfter) {
-        if (spaceAfter && StringUtils.isNullOrEmpty(container.getAfter().getWhitespace())) {
+        if (!container.getAfter().getComments().isEmpty()) {
+            return container;
+        }
+
+        if (spaceAfter && notSingleSpace(container.getAfter().getWhitespace())) {
             return container.withAfter(container.getAfter().withWhitespace(" "));
-        } else if (!spaceAfter && container.getAfter().getWhitespace().equals(" ") && container.getAfter().getComments().isEmpty()) {
+        } else if (!spaceAfter && onlySpacesAndNotEmpty(container.getAfter().getWhitespace())) {
             return container.withAfter(container.getAfter().withWhitespace(""));
         } else {
             return container;
         }
     }
 
+    /**
+     * Checks if a string only contains spaces (excluding newline characters).
+     * @return true if contains spaces only, or true for empty string.
+     */
+    private static boolean onlySpaces(String str) {
+        for (int i = 0; i < str.length(); i++) {
+            if (str.charAt(i) != ' ') {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
Don't forget about tabs

Reply:
good point, I actually considered the tabs, and I found [TabsAndIndentsVisitor](https://github.com/openrewrite/rewrite/blob/e9b98561ad03f5bb95a0b3feb786243dd45d9847/rewrite-java/src/main/java/org/openrewrite/java/format/TabsAndIndentsVisitor.java) has some handling of `\t`. so I was concerned about it and decided to limit it to whitespace only.

Reply:
revised to include tabs.